pkg/apis/servicecatalog/validation/instance_test.go

```diff
@@ -141,3 +142,124 @@ func TestValidateInstance(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateInstanceUpdate(t *testing.T) {
```

Comment:
Since this is run on every status update, we need to check things like transitions in conditions, etc.  Right now I believe the code will disallow updating conditions while an async op is in progress unless they are updated in a status update that sets that field back to false.  I think those updates should be allowed, because you could conceivably have error conditions change during that time - imagine a broker that goes down during an async deprovision  - you would likely have a message on the ready condition change, but not update that value of the async op field.